libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1185,6 +1186,188 @@ string YulUtilFunctions::clearStorageStructFunction(StructType const& _type)
 	});
 }
 
+string YulUtilFunctions::copyArrayFromMemoryToStorage(ArrayType const& _fromType, ArrayType const& _toType)
+{
+	solAssert(_fromType.location() == DataLocation::Memory, "");
+
+	string functionName = "copy_array_memory_to_storage_from_" + _fromType.identifier() + "_to_" + _toType.identifier();
+	return m_functionCollector.createFunction(functionName, [&](){
+		Whiskers templ(R"(
+			function <functionName>(slot, memptr) {
+				let length := <arrayLength>(memptr)
+				<?isToDynamic>
+					<resizeArray>(slot, length)
+				</isToDynamic>
+
+				let memoryOffset := <srcDataLocation>(memptr)
+
+				let elementSlot := <dstDataLocation>(slot)
+				<?isBytes>
+					if lt(length, 32) {
+						elementSlot := slot
+					}
+				</isBytes>
+				let elementOffset := 0
+
+				for { let i := 0 } lt(i, length) {i := add(i, 1)} {
```

Comment:
We could unroll the loop to "number of items per slot"

Reply:
I can do it. If size of type is less than 32 which is only for value types, I can do all operation on whole slot and type u256?

Reply:
I'll leave this as is now and will do it as a part of #9831, ok?